code/game/objects/items.dm

```diff
@@ -28,6 +28,8 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
 	///Icon file for right inhand overlays
 	var/righthand_file = 'icons/mob/inhands/items_righthand.dmi'
 
+	///Override mob worn overlay with invisible overlay. Default is true to allow for a large number of items that do not have an icon when worn, particularly in the belt or suit, to remain invisible instead of displaying a missing texture sprite.
+	var/worn_hidden = TRUE

```

Comment:
No reason to have a code solution for this when you can just delete the "no_name" / missing texture icon instead to do the same thing. This is just band-aiding over the issue.

Reply:
I lean more towards removing this var and just adding the remaining missing sprites now that most of them are fixed instead of removing the missing texture sprite myself. It's pointed out a good number of bugs so far that have been (once I noticed) mostly resolved at this point.

Reply:
The point of having a "invisible" sprite is to mark them as explicitly set to not have a sprite, whereas before it could have just been an oversight

Reply:
Sorry for the delay @ShizCalev, I'll take a look at making changes and resolving the conflicts here in a couple of days. I'll have a couple more item's worn_icon_state fixed as well.

For concealable items, I suppose we could define the worn_icon and worn_icon_state to the hidden icon instead of having a worn_hidden var, but my thought is it would be easier using the worn_hidden bool.

As for making items hidden by default using worn_hidden, my reasoning is we can suppress the items in the list below from showing the missing texture sprite from production servers in the meantime as sprites are being made for them since the list should now comprehensively identify the items needing sprites. The items could then be unhidden as they receive sprites, though I can see where this might be more trouble than it's worth especially if others were unaware of the change.

This info found by tracking:
ITEM_SLOT_BELT
ITEM_SLOT_BACK
/obj/item/clothing/suit/*    allowed = list(...)

**mask.dmi**
- /obj/item/instrument/harmonica

**back.dmi**
- /obj/item/bikehorn
- /obj/item/bikehorn/airhorn
- /obj/item/bikehorn/golden
- /obj/item/bikehorn/rubberducky
- /obj/item/bikehorn/rubberducky/plasticducky
- /obj/item/nullrod/pride_hammer
- /obj/item/singularityhammer
- /obj/item/mjollnir
- /obj/item/kinetic_crusher
- /obj/item/gun/magic/staff 
- /obj/item/gun/magic/staff/locker
- /obj/item/gun/magic/staff/flying
- /obj/item/gun/magic/staff/sapping
- /obj/item/gun/magic/staff/necropotence
- /obj/item/gun/magic/staff/wipe
- /obj/item/shield/adamantineshield

**belt.dmi**
- /obj/item/bikehorn
- /obj/item/bikehorn/airhorn
- /obj/item/bikehorn/golden
- /obj/item/bikehorn/rubberducky
- /obj/item/bikehorn/rubberducky/plasticducky
- /obj/item/airlock_painter
- /obj/item/airlock_painter/decal
- /obj/item/airlock_painter/decal/debug
- /obj/item/binoculars
- /obj/item/lighter
- /obj/item/lighter/greyscale
- /obj/item/lighter/slime
- /obj/item/flashlight
- /obj/item/flashlight/lamp
- /obj/item/flashlight/lamp/green
- /obj/item/flashlight/lamp/bananalamp
- /obj/item/flashlight/flare
- /obj/item/flashlight/latern
- /obj/item/flashlight/latern/heirloom_moth
- /obj/item/flashlight/latern/jade
- /obj/item/flashlight/latern/syndicate
- /obj/item/flashlight/slime
- /obj/item/flashlight/emp
- /obj/item/flashlight/emp/debug
- /obj/item/flashlight/glowstick
- /obj/item/flashlight/glowstick/red
- /obj/item/flashlight/glowstick/blue
- /obj/item/flashlight/glowstick/cyan
- /obj/item/flashlight/glowstick/orange
- /obj/item/flashlight/glowstick/yellow
- /obj/item/flashlight/glowstick/pink
- /obj/item/flashlight/flashdark
- /obj/item/flashlight/eyelight
- /obj/item/geiger_counter
- /obj/item/geiger_counter/cyborg
- /obj/item/lightreplacer
- /obj/item/lightreplacer/cyborg
- /obj/item/radio
- /obj/item/radio/off
- /obj/item/radio/borg
- /obj/item/radio/borg/syndicate
- /obj/item/nanite_scanner
- /obj/item/grenade
- /obj/item/grenade/antigravity
- /obj/item/grenade/barrier
- /obj/item/grenade/chem_grenade
- /obj/item/grenade/chem_grenade/*
- /obj/item/grenade/clusterbuster
- /obj/item/grenade/clusterbuster/*
- /obj/item/grenade/flashbang
- /obj/item/grenade/flashbang/cluster
- /obj/item/grenade/empgrenade
- /obj/item/grenade/firecracker
- /obj/item/grenade/stingbang
- /obj/item/grenade/stingbang/*
- /obj/item/grenade/primer
- /obj/item/grenade/primer/stingbang
- /obj/item/grenade/iedcasing
- /obj/item/grenade/iedcasing/spawned
- /obj/item/grenade/hypnotic
- /obj/item/grenade/c4
- /obj/item/grenade/c4/x4
- /obj/item/grenade/smokebomb
- /obj/item/grenade/spawnergrenade
- /obj/item/grenade/spawnergrenade/*
- /obj/item/grenade/syndieminibomb
- /obj/item/grenade/syndieminibomb/concussion
- /obj/item/grenade/frag
- /obj/item/grenade/frag/mega
- /obj/item/grenade/gluon
- /obj/item/nullrod/claymore/darkblade
- /obj/item/nullrod/claymore/chainsaw_sword
- /obj/item/nullrod/claymore/glowing
- /obj/item/nullrod/claymore/multiverse
- /obj/item/nullrod/claymore/saber
- /obj/item/nullrod/claymore/saber/red
- /obj/item/nullrod/claymore/saber/pirate
- /obj/item/nullrod/sord
- /obj/item/sord
- /obj/item/nullrod/scythe/talking/chainsword
- /obj/item/nullrod/hammmer
- /obj/item/nullrod/whip
- /obj/item/melee/curator_whip
- /obj/item/melee/chainofcommand
- /obj/item/melee/chainofcommand/tailwhip
- /obj/item/melee/chainofcommand/tailwhip/kitty
- /obj/item/key/lasso
- /obj/item/nullrod/hypertool
- /obj/item/nullrod/spear
- /obj/item/melee/classic_baton/telescopic
- /obj/item/melee/classic_baton/telescopic/contractor_baton
- /obj/item/melee/roastingstick
- /obj/item/melee/cleric_mace
- /obj/item/construction/rcd
- /obj/item/construction/rcd/*
- /obj/item/construction/rld
- /obj/item/construction/rld/mini
- /obj/item/pipe_dispenser
- /obj/item/storage/bag
- /obj/item/storage/bag/*
- /obj/item/storage/box/matches
- /obj/item/storage/fancy/candle_box
- /obj/item/storage/fancy/cigarettes
- /obj/item/storage/fancy/cigarettes/cigars
- /obj/item/storage/fancy/cigarettes/cigars/*
- /obj/item/storage/fancy/cigarettes/cigpack_*
- /obj/item/storage/fancy/cigarettes/dromedaryco
- /obj/item/banhammer
- /obj/item/soulstone
- /obj/item/soulstone/*
- /obj/item/secateurs
- /obj/item/geneshears
- /obj/item/shovel
- /obj/item/shovel/serrated
- /obj/item/shovel/spade
- /obj/item/melee/transforming/cleaving_saw
- /obj/item/melee/transforming/cleaving_saw/miner
- /obj/item/modular_computer/tablet
- /obj/item/modular_computer/tablet/*
- /obj/item/clipboard
- /obj/item/ammo_casing
- /obj/item/ammo_casing/*
- /obj/item/ammo_box
- /obj/item/ammo_box/*
- /obj/item/reagent_containers/hypospray
- /obj/item/reagent_containers/hypospray/medipen
- /obj/item/reagent_containers/hypospray/medipen/*
- /obj/item/reagent_containers/hypospray/combat
- /obj/item/reagent_containers/hypospray/combat/nanites
- /obj/item/reagent_containers/hypospray/combat/heresypurge
- /obj/item/reagent_containers/hypospray/CMO
- /obj/item/reagent_containers/medigel
- /obj/item/reagent_containers/medigel/*
- /obj/item/reagent_containers/spray
- /obj/item/reagent_containers/spray/*
- /obj/item/destTagger
- /obj/item/destTagger/borg
- /obj/item/sales_tagger
- /obj/item/uplink
- /obj/item/uplink/*

**belt_mirror.dmi**
- /obj/item/bikehorn
- /obj/item/bikehorn/airhorn
- /obj/item/bikehorn/golden
- /obj/item/bikehorn/rubberducky
- /obj/item/bikehorn/rubberducky/plasticducky
- /obj/item/instrument
- /obj/item/instrument/*
- /obj/item/reagent_containers/food/snacks/pie/cream
- /obj/item/reagent_containers/food/snacks/grown/banana
- /obj/item/grown/bananapeel
- /obj/item/megaphone/clown
- /obj/item/soap
- /obj/item/soap/*
- /obj/item/reagent_containers/food/drinks/soda_cans/canned_laughter
- /obj/item/toy/crayon
- /obj/item/toy/crayon/spraycan
- /obj/item/toy/crayon/spraycan/lubecan
- /obj/item/analyzer
- /obj/item/tank/internals/plasma
- /obj/item/storage/book/bible
- /obj/item/storage/book/bible/booze
- /obj/item/storage/book/bible/syndicate
- /obj/item/nullrod
- /obj/item/nullrod/* (w_class<5)
- /obj/item/reagent_containers/food/drinks/bottle/holywater
- /obj/item/reagent_containers/food/drinks/bottle/holywater/hell
- /obj/item/storage/fancy/candle_box
- /obj/item/candle
- /obj/item/candle/infinite
- /obj/item/flashlight
- /obj/item/flashlight/*
- /obj/item/tome
- /obj/item/melee/cultblade
- /obj/item/melee/cultblade/dagger
- /obj/item/melee/cultblade/ghost
- /obj/item/melee/sickly_blade
- /obj/item/melee/sickly_blade/*
- /obj/item/forbidden_book
- /obj/item/forbidden_book/debug
- /obj/item/construction/rcd
- /obj/item/pipe_dispenser
- /obj/item/resonator
- /obj/item/storage/bag/ore
- /obj/item/storage/bag/ore/*
- /obj/item/ammo_casing
- /obj/item/ammo_casing/*
- /obj/item/ammo_box
- /obj/item/ammo_box/*
- /obj/item/melee/transforming/energy/sword
- /obj/item/melee/transforming/energy/sword/*
- /obj/item/clothing/glasses/eyepatch
- /obj/item/reagent_containers/food/drinks/bottle/rum
- /obj/item/stack/spacecash
- /obj/item/stack/spacecash/*
- /obj/item/teleportation_scroll
- /obj/item/teleportation_scroll/apprentice
- /obj/item/storage/firstaid
- /obj/item/storage/firstaid/*
- /obj/item/stack/medical
- /obj/item/stack/medical/*
- /obj/item/circular_saw
- /obj/item/storage/box/pillbottles
- /obj/item/reagent_containers/pill
- /obj/item/reagent_containers/pill/*
- /obj/item/storage/pill_bottle
- /obj/item/storage/pill_bottle/*
- /obj/item/storage/bag/chemistry
- /obj/item/reagent_containers/glass/beaker
- /obj/item/reagent_containers/glass/beaker/*
- /obj/item/reagent_containers/glass/bottle
- /obj/item/reagent_containers/glass/bottle/*
- /obj/item/reagent_containers/glass/bucket
- /obj/item/reagent_containers/glass/bucket/*
- /obj/item/reagent_containers/dropper
- /obj/item/reagent_containers/syringe
- /obj/item/reagent_containers/syringe/*
- /obj/item/reagent_containers/hypospray
- /obj/item/reagent_containers/hypospray/*
- /obj/item/reagent_containers/spray/chemsprayer/janitor
- /obj/item/reagent_containers/spray/pestspray
- /obj/item/reagent_containers/spray/plantbgone
- /obj/item/kitchen/knife/combat
- /obj/item/kitchen/knife/combat/*
- /obj/item/multitool
- /obj/item/multitool/*
- /obj/item/storage/bag/trash
- /obj/item/storage/bag/trash/*
- /obj/item/melee/flyswatter
- /obj/item/mop
- /obj/item/mop/*
- /obj/item/holosign_creator
- /obj/item/holosign_creator/*
- /obj/item/pneumatic_cannon/speargun
- /obj/item/banner
- /obj/item/banner/*
- /obj/item/spear
- /obj/item/spear/*
- /obj/item/organ/regenerative_core/legion
- /obj/item/plant_analyzer
- /obj/item/seeds
- /obj/item/seeds/*
- /obj/item/cultivator
- /obj/item/cultivator/rake
- /obj/item/hatchet
- /obj/item/hatchet/*
- /obj/item/storage/bag/plants
- /obj/item/storage/bag/plants/portaseeder
- /obj/item/graft
- /obj/item/secateurs
- /obj/item/geneshears
- /obj/item/kitchen
- /obj/item/kitchen/*
- /obj/item/radio
- /obj/item/radio/*
- /obj/item/scalpel
- /obj/item/surgical_drapes
- /obj/item/cautery
- /obj/item/hemostat
- /obj/item/retractor
- /obj/item/melee/curator_whip
- /obj/item/paper
- /obj/item/paper/*
- /obj/item/melee/classic_baton/telescopic
- /obj/item/toy
- /obj/item/toy/*
- /obj/item/clothing/mask/facehugger/toy
- /obj/item/spellbook
- /obj/item/disk
- /obj/item/stamp
- /obj/item/reagent_containers/food/drinks/flask
- /obj/item/melee
- /obj/item/melee/*
- /obj/item/storage/lockbox/medal
- /obj/item/storage/lockbox/medal/*
- /obj/item/assembly/flash/handheld
- /obj/item/storage/box/matches
- /obj/item/lighter
- /obj/item/lighter/greyscale
- /obj/item/lighter/slime
- /obj/item/clothing/mask/cigarette
- /obj/item/storage/fancy/cigarettes

Reply:
oh amazing, saves me a fuckton of time looking for everything. throw that into an issue report and i'll widdle that down over the next week.

Reply:
Set worn_hidden to FALSE by default, but still can be used to explicitly set items as hidden as optimumtact mentioned. Worn_hidden is currently being used to fix neckerchiefs as neckerchiefs in of themselves are invisible, but generate an overlay using the underlying bandana mask item, moving the overlay down a few pixels, and changing the layer level. Also fixed a couple more item's worn_icon_state. Let me know what you think @ShizCalev.

Reply:
Took out var and fixed neckerchief underlying layer with direct input of a hidden icon.